api/src/main/scala/org/ensime/api/outgoing.scala

```diff
@@ -235,8 +235,7 @@ final case class EmptySourcePosition() extends SourcePosition
 final case class OffsetSourcePosition(file: EnsimeFile, offset: Int) extends SourcePosition
 final case class LineSourcePosition(file: EnsimeFile, line: Int) extends SourcePosition
 
-case class SourceHint(position: SourcePosition, preview: Option[String])
-case class SourceHints(sourceHints: List[SourceHint]) extends RpcResponse
+final case class SourcePositions(positions: List[SourcePosition], previews: List[String]) extends RpcResponse
```

Comment:
you could just leave the field called `positions` and change the type. This is a bit messy.